### PR TITLE
Start Analysis VC

### PR DESCRIPTION
**Pull request by tomalley104**

### CODE REVIEW

**Comment by tomalley104** [comment]
I think that the only thing yet to be touched on was:

`if let error = result.error as? SpotifyError { ... }` 

Last thing I said was this is because I'm using Intrepid's Result enum that has `ErrorType` associated with `.Failure`. Is there a way around this outside of just rewriting the enum? Will the compiler even let me do that?

**Comment by brightredchilli** [comment]
hmmm....I see why now. I think it might be helpful to make the error parameter generic as well, but that's outside of scope for this pr, we can leave it as it is.

File: spotifyMini/Models/Analysis.swift [hunk 1]
```diff
@@ -8,9 +8,68 @@
...
+    // MARK: Fetching
+
+    func fetchAnalyses() {
+        let ids = self.tracks.map { $0.identifier } as [String]
+        self.spotify.fetchAnalysis(forTrackIDs: ids) { result in
+            self.trackAnalyses = result.value
+            var userInfo: [String:AnyObject]?
+            if let error = result.error as? SpotifyError {
+                userInfo = [String:AnyObject]()
+                userInfo?.updateValue(NSError(spotifyError: error), forKey: AnalysisNotificationErrorUserInfoKey)
```
Comment (brightredchilli) [inline spotifyMini/Models/Analysis.swift]: cleaner here like so `userInfo = [AnalysisNotificationErrorUserInfoKey : NSError(spotifyError: error) ]`

File: spotifyMini/Models/Artist.swift [hunk 2]
```diff
@@ -61,6 +62,7 @@ class Artist {
                 self.topTracks = result.value
                 var userInfo: [String:AnyObject]?
                 if let error = result.error as? SpotifyError {
+                    userInfo = [String:AnyObject]()
```
Comment (brightredchilli) [inline spotifyMini/Models/Artist.swift]: seeing that this line is repeated over and over again, you should find a way to consolidate. Consider making an NSNotificationCenter or NSNotification extension in another PR

**Comment by brightredchilli** [comment]
:ship:

**Pull request merged**